README.rst

```diff
@@ -0,0 +1,226 @@
+    enabled for internal computations by providing suitable
+    serialisation/deserialisation functions.
+
+- **Failure handling**. In a long running workflow, failures are inevitable. FunFlow
+  has support for handling failures inside workflows, and for resuming workflows
+  from the last successful point once some external error has been corrected.
+- Sequential and **parallel execution**. FunFlow executes sections of workflows
+  in parallel where this is possible, and handles sequential execution where
+  tasks have dependencies.
+- **Task distribution**. External steps can be serialised and run remotely. FunFlow
+  includes a number of central coordinators which handle distributing steps
+  among multiple machines:
+
+  - An `in-memory <./funflow/src/Control/FunFlow/External/Coordinator/Memory.hs>`_
+    coordinator, useful for single-process computation.
+  - A `Redis based <./funflow/src/Control/FunFlow/External/Coordinator/Redis.hs>`_
+    coordinator.
+  - A `SQLite <./funflow/src/Control/FunFlow/External/Coordinator/SQLite.hs>`_
+    coordinator, which uses a SQLite database as a shared location for task
+    distribution.
+
+- Support for distributing complete workflows, rather than just individual
+  steps. The `funflow-jobs <./funflow-jobs>`_ package provides support for
+  distributing full workflows using Redis. This has some extra constraints
+  compared to distributing external tasks, since Haskell functions are in
+  general not serialisable.
+- Safe and powerful workflow composition. FunFlow takes advantage of Haskell's
+  type system to ensure flows are composed in a safe manner. And since the
+  composition of flows results in a flow, it's easy to include one complex
+  workflow as part of another.
+
+Defining a workflow
+-------------------
+
+A simple workflow can be defined from basic Haskell functions::
+
+  -- | A flow which doubles and reverses a list
+  doubleAndReverse :: SimpleFlow [Int] [Int]
+  doubleAndReverse = mapA (step (*2)) >>> step reverse
+
+The `step` function takes a Haskell function of type `a -> b` and lifts it into
+a `SimpleFlow a b`. [1]_ The `mapA` combinator applies a flow over a list, and
+`>>>` composes two flows sequentially.
+
+We can also use Haskell's arrow notation to compose more complex flows::
+
+  -- | Flow which asks for the user's name, then their favourite number.
+  --   The favourite number will be cached, such that if the user comes back,
+  --   they will not be asked again.
+  favouriteNumber = proc () -> do
+    name <- stepIO (\_ -> putStrLn "Hi, what's your name?" >> const getLine) -< ()
+    fnum <- stepIO' (def { cache = defaultCacherWithIdent 172652 })
+                    (\_ -> putStrLn "What's your favourite number?" >> const getLine) -< name
+    returnA -< (name, fnum)
+
+In this flow, we can also see the use of FunFlow's caching. `stepIO` lets us lift
+an `IO` action into a flow. `stepIO'` is a more complex variant that lets us configure
```

Comment:
I'm never keen on "lets you do something" - it has the kind of odd sense of it giving you permission. Normally would replace with just the straightforward what-you-use-it-for - eg "(You can) use the more complex variant `stepIO` to configure things like the caching properties".